core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -119,6 +125,10 @@ public synchronized ChannelImpl shutdown() {
       return this;
     }
     shutdown = true;
+    if (deadlineCancellationExecutor != null) {
+      deadlineCancellationExecutor =
```

Comment:
This should probably happen when `terminated = true` below. That will help guarantee it only happens once (since elsewhere release code you added will run when `shutdown == true` but not when `terminated == true`)